lib/runners/processor/actionlint.rb

```diff
@@ -0,0 +1,87 @@
+module Runners
+  class Processor::Actionlint < Processor
+    SCHEMA = _ = StrongJSON.new do
+      extend Schema::ConfigTypes
+
+      # @type self: SchemaClass
+      let :config, base(
+        target: target,
+
+        # Native options
+        ignore: one_or_more_strings?,
+      )
+    end
+
+    ALLOWED_ERRORS = [
+      "no project was found in any parent directories of",
+      "no YAML file was found in",
+    ].freeze
+
+    register_config_schema SCHEMA.config
+
+    def self.config_example
+      <<~'YAML'
+        root_dir: project/
+        target:
+          - .github/workflows/one_workflow.yml
+          - .github/workflows/another_workflow.yml
+        ignore:
+          - "job .* needs job"
+          - "unknown Webhook event"
+      YAML
+    end
+
+    # This analyser uses git metadata (.git/).
+    def use_git_metadata_dir?
+      true
+    end
+
+    def analyzer_options
+      [].tap do |opts|
+        opts << "-format" << "{{json .}}"
+        # Specify error message patterns to ignore
+        Array(config_linter[:ignore]).each { |v| opts << "-ignore" << v }
+        # After -- separator, specify target files to analyze
+        opts << "--"
+        Array(config_linter[:target]).each { |v| opts << v }
+      end
+    end
+
+    def analyze(changes)
+      stdout, stderr, status = capture3(
+        analyzer_bin,
+        *analyzer_options,
+      )
+
+      # No yaml files found => Success. Other outcomes (e.g. fatal errors) with exit code 2, 3 => Failure.
+      if ALLOWED_ERRORS.any? { |pattern| stderr.include?(pattern) }
+        add_warning(stderr)
+        return Results::Success.new(guid: guid, analyzer: analyzer)
+      elsif not [0, 1].include?(status.exitstatus)
+        add_warning(stderr)
+        return Results::Failure.new(guid: guid, analyzer: analyzer)
+      end
+
+      # If zero or more issues were found, consider it success
+      Results::Success.new(guid: guid, analyzer: analyzer).tap do |result|
+        parse_result(stdout) do |issue|
+          result.add_issue issue
+        end
+      end
+    end
+
+    def parse_result(output)
+      json = JSON.parse(output, symbolize_names: true)
+      json.flat_map do |issue|
+        # NOTE: the actionlint JSON output contains a `issue[:snippet]` field as well,
+        # however Sider has no use for it at the moment.
+        yield Issue.new(
+          id: issue[:kind],
+          path: relative_path(issue[:filepath]),
+          location: Location.new(start_line: issue[:line], start_column: issue[:column]),
+          message: issue[:message],
```

Comment:
[suggestion] How about adding a ShellCheck link like [SC2221](https://github.com/koalaman/shellcheck/wiki/SC2221)?

```suggestion
          message: issue[:message],
          links: shellcheck_issue_id ? ["#{analyzer_github(:shellcheck)}/wiki/#{shellcheck_issue_id}"] : [],
```

https://github.com/sider/runners/blob/f308d1924678b644318a906010b3038b8b848985/lib/runners/processor/shellcheck.rb#L159

Reply:
In addition, saving a ShellCheck issue ID seems good like this:

```ruby
object: {
  shellcheck_issue_id: shellcheck_issue_id,
},
schema: SCHEMA.issue,
```

Reply:
Let me try parsing out the shellcheck issue code as you suggested.

Reply:
Added ShellCheck issue ID parsing.